social_django/storage.py

```diff
@@ -145,6 +145,51 @@ def create_social_auth(cls, user, uid, provider):
         return social_auth
 
 
+class CompliantDjangoUserMixin(DjangoUserMixin):
+    @property
+    def access_token(self):
+        """Override method in UserMixin as we've broken it out of extra_data"""
+        return self.actual_access_token
+
+    def refresh_token(self, strategy, *args, **kwargs):
+        """Override method in UserMixin as tokens are in their own fields now"""
+        token = self.actual_refresh_token or self.actual_access_token
+        backend = self.get_backend_instance(strategy)
+        if token and backend and hasattr(backend, 'refresh_token'):
+            response = backend.refresh_token(token, *args, **kwargs)
+            extra_data = backend.extra_data(self,
+                                            self.uid,
+                                            response,
+                                            self.extra_data)
+            # break the access token and refresh token out of the extra data
+            self.actual_access_token = extra_data.pop('access_token', None)
+            self.actual_refresh_token = extra_data.pop('refresh_token', None)
+            if self.set_extra_data(extra_data):
+                self.save()
+
+    def set_extra_data(self, extra_data=None):
+        """
+        Making sure we never store the tokens in extra data
+        """
+        access_token = extra_data.pop('access_token', None)
```

Comment:
What happens if extra_data is None here?

Reply:
We just set it to None!

Reply:
Will None.pop(..)  return None or create an error though?

Reply:
Oh wait I misread the question, good point